packages/driver/src/dom/elements/complexElements.ts

```diff
@@ -161,6 +161,10 @@ export const isFocusedOrInFocused = (el: HTMLElement) => {
 
   debug('elToCheckCurrentlyFocused', elToCheckCurrentlyFocused)
 
+  while (elToCheckCurrentlyFocused !== activeElement && activeElement?.shadowRoot?.activeElement) {
```

Comment:
Is it possible to check to check if the element is within the shadow root before pulling out `doc` on line 143 like we do with `isFocused()` here https://github.com/cypress-io/cypress/blob/develop/packages/driver/src/dom/elements/complexElements.ts#L112?
```
    let doc
    if (isWithinShadowRoot(el)) {
      doc = el.getRootNode()
    } else {
      doc = $document.getDocumentFromElement(el)
    }
```

Reply:
Good call; updated.